Mathlib/Topology/Spectral/Prespectral.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Order.Ideal
+import Mathlib.Topology.Sets.Compacts
+import Mathlib.Topology.Sets.OpenCover
+import Mathlib.Topology.Spectral.Hom
+
+/-!
+
+# Prespectral spaces
+
+In this file, we define prespectral spaces as spaces whose lattice of compact opens forms a basis.
+
+-/
+
+open TopologicalSpace Topology
+
+variable {X Y : Type*} [TopologicalSpace X] [TopologicalSpace Y]
+
+/-- A space is prespectral if the lattice of compact opens forms a basis. -/
+@[mk_iff]
+class PrespectralSpace (X : Type*) [TopologicalSpace X] : Prop where
+  isTopologicalBasis : TopologicalSpace.IsTopologicalBasis { U : Set X | IsOpen U ∧ IsCompact U }
+
+/-- A space is prespectral if it has a basis consisting of compact opens. -/
+lemma PrespectralSpace.of_isTopologicalBasis {B : Set (Set X)}
+    (basis : IsTopologicalBasis B) (isCompact_basis : ∀ U ∈ B, IsCompact U) :
+    PrespectralSpace X where
+  isTopologicalBasis.exists_subset_inter _ ht₁ _ ht₂ _ hx :=
+    have ⟨v, hvB, hv⟩ := basis.exists_subset_of_mem_open hx (ht₁.1.inter ht₂.1)
+    ⟨v, ⟨basis.isOpen hvB, isCompact_basis v hvB⟩, hv⟩
+  isTopologicalBasis.sUnion_eq := Set.univ_subset_iff.mp (basis.sUnion_eq.superset.trans
+    (Set.sUnion_mono fun s hs ↦ ⟨basis.isOpen hs, isCompact_basis s hs⟩))
+  isTopologicalBasis.eq_generateFrom := (le_generateFrom fun _ h ↦ h.1).antisymm (le_trans
+    (generateFrom_anti fun s hs ↦ ⟨basis.isOpen hs, isCompact_basis s hs⟩) basis.eq_generateFrom.ge)
+
+/-- A variant with an indexed basis instead. -/
```

Comment:
```suggestion
/-- A space is prespectral if it has an indexed basis consisting of compact opens. -/
```
I don't like docstrings that are ununderstandable on hover

Reply:
The point of this docstring is to point out the difference from the unprimed one. The docstring shouldn't just repeat the type signature (which is visible on hover) IMO.

But in this case the docstring you proposed is short enough that I am happy to add it alongside the existing one.

Reply:
What I am saying is that if you were to hover over that lemma and read the "variant", you wouldn't know what this is supposed to be a variant of.

Reply:
I think it is a naming consensus that the primed lemma is a variant of the unprimed lemma?

Reply:
Anyways does the current docstring look fine to you?

Reply:
1. Then why add a docstring saying this?
2. I believe a significant proportion of primed lemmas are primed not because they are a variant of the unprime lemma, but because there is a naming clash

Reply:
As I said above, the point is to point out the difference from the unprimed one (i.e. "indexed basis"). I think anything else is just noise.

But is the current docstring okay? I'm satisfied with it.

Reply:
Yeah, I'm happy with it now :)